api/v1alpha3/azuremachine_validation.go

```diff
@@ -165,6 +174,26 @@ func ValidateManagedDisk(old, new ManagedDisk, fieldPath *field.Path) field.Erro
 	return allErrs
 }
 
+// ValidateManagedDisks validates updates to Data disks' managed disk field.
+func ValidateManagedDisks(old, new []DataDisk, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(old) != len(new) {
```

Comment:
@CecileRobertMichon I am not sure if this check is required anymore. Can you clarify what changes are allowed in terms of updates to data disks?  Can a user add/remove data disks? what specific fields for data disks can a user update vs which fields are read only?

Reply:
Currently, a user connot add/remove/modify data disks on an existing machine. Validation should reflect that.

Reply:
Thank you for the clarification.